### PR TITLE
rc_input: move to PX4 WQ and minor cleanup

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

File: src/drivers/rc_input/RCInput.cpp [hunk 1]
```diff
@@ -86,32 +78,23 @@ RCInput::~RCInput()
...
 	_rcs_fd = dsm_init(_device);
 
 	if (_rcs_fd < 0) {
 		return -errno;
 	}
 
-	if (board_rc_swap_rxtx(_device)) {
-		ioctl(_rcs_fd, TIOCSSWAP, SER_SWAP_ENABLED);
-	}
+	ioctl(_rcs_fd, TIOCSSWAP, SER_SWAP_ENABLED);
```
Comment (dagar) [inline src/drivers/rc_input/RCInput.cpp]: Is this even needed? RC_SERIAL_SWAP_RXTX doesn't seem to be used anywhere.

**Comment by dagar** [comment]
@bkueng still interested in polling rc?

**Comment by bkueng** [comment]
> @bkueng still interested in polling rc?

Yes

File: src/drivers/rc_input/RCInput.cpp (hunk 1 above)
Comment (dagar) [reply src/drivers/rc_input/RCInput.cpp]: Talked to @davids5, this is safe to keep.

**Comment by dannyfpv** [comment]
tested on pixhawk4 v5 f-450

**Modes Tested**

Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.
Mission Plan Mode (Automated): Good.
RTL: Good.

**- Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoin activate RTL and see landing behaviour.

**Notes:**
No issues noted, good flight in general.

**Log:**
https://review.px4.io/plot_app?log=24bc92c2-8e7f-47c9-abfa-b40d8db75237

**Comment by Junkim3DR** [comment]
### Tested on NXP FMUK66 v3

**Modes Tested**
- Position Mode: Good.
- Altitude Mode: Good.
- Stabilized Mode: Good.
- Mission Plan Mode (Automated): Good.
-RTL (Return To Land): Good.

**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode proceed to switch to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint then trigger RTL and see landing behavior.

**Notes**
No issues noted, good flight in general.

**Logs**
https://review.px4.io/plot_app?log=da2b13d6-8452-4115-9139-f6e0835d1f77

**Comment by jorge789** [comment]
### Tested on PixRacer V4:                            
                                                                
**Flight Card 1**

**Modes Tested:**
Position Mode: Good.
Altitude Mode: Good.
Stabilized Mode: Good.

**Procedure:**
Arm and Take off in position mode, after flying for approximately one minute, switched to altitude then stabilized mode.
**Notes:**
No issues were noted, good flight in general.

**Logs:**  https://review.px4.io/plot_app?log=d6ca40d9-91fe-457a-bb50-595f0cd26710

**Flight Card 2**

**Modes Tested**
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**

Armed form QGC.
Took-off on mission plan mode
. Note that commands were sent form QGC (Armed and takeoff)
Once all waypoint completed vehicle switched to RTL and landed as expected.
Good flight in general.
Note:
No issues were noted, good flight in general.

**Logs:**  https://review.px4.io/plot_app?log=e89d3edf-df09-4175-b146-9eab9748a002

**Flight Card 3**
Modes Tested
Position Mode: Good.
Mission Plan Mode (Automated): Good.
RTL (Return To Land): Good.

**Procedure**
Arm and Take off in position mode, after flying for approximately one minute, switched to mission plan mode then make sure that vehicle follows all waypoints as shown in QGC, once completed all waypoint see landing behavior.
Good flight in general.

**Note:**
No issues were noted, good flight in general.

**Logs:** https://review.px4.io/plot_app?log=67c7e2e4-2ead-4317-84a2-348e41178139

**Flight Card 4**
**Modes Tested**
Fail safe:The radio control turned off and the vehicle returned and landed correctly.
Tested good.

**Logs:**  https://review.px4.io/plot_app?log=75ae478f-eaf5-442c-bfcc-eabcb4e480a8

**Pull request merged**